src/pages/collab/index.tsx

```diff
@@ -0,0 +1,287 @@
+import { useEffect, useRef, useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+
+import { api } from "~/utils/api";
+
+import LoadingIcon from "~/components/LoadingIcon";
+
+const MatchRequestPage = () => {
+  const [pageState, setPageState] = useState({
+    difficulty: -1,
+    category: "",
+    id: "",
+    response: "",
+    statusMessage: "Searching for partner...",
+    isWaiting: false,
+    requestFailed: false,
+    hasSetDifficulty: false,
```

Comment:
I think we can do away with these fields? The same check can be done by
```ts
difficulty != -1 && category != ""

```

Reply:
the hasSet fields are for the error messages, they only appear if the user hasnt selected/put any value for the field yet and the user clicks on the match button, which will set these fields to true and any change to the inputs before or after can toggle the hasSet fields

Reply:
If we need such a functionality we could make do with just one `hasSubmitted` field to do the check?
Could also considering using select element instead of just  a div as we can set the required flag so we don't have to handle this kind of logic
https://www.w3schools.com/tags/att_select_required.asp

Reply:
True, I refactored to hasSubmitted, I think the div is better for styling purposes but we can change it later